packages/wonder-blocks-banner/src/components/banner.js

```diff
@@ -0,0 +1,281 @@
+     */
+    | "success"
+    /**
+     * Color gold, triangle exclamation-point icon
+     */
+    | "warning"
+    /**
+     * Color red, circle exclamation-point icon
+     */
+    | "critical";
+
+type BannerLayout =
+    /**
+     * Renders a rounded rectangle, usually for when banner is used as an inset
+     * element on a screen (e.g., the SOT card) that appears to be floating
+     */
+    | "floating"
+    /**
+     * Renders a full-width banner, with no rounded corners.
+     * This is the default.
+     */
+    | "full-width";
+
+type ActionTrigger = {|
+    title: string,
+    onClick?: () => void,
+    href?: string,
+    ariaLabel?: string,
+|};
+
+type Props = {|
+    /**
+     * Determines the color and icon of the banner.
+     */
+    kind?: BannerKind,
+
+    /**
+     * Determines the edge style of the Banner.
+     */
+    layout?: BannerLayout,
+
+    /**
+     * Text on the banner (LabelSmall) or a node if you want something different
+     */
+    text: string | React.Node,
+
+    /**
+     * Links or tertiary Buttons that appear to the right of the text.
+     *
+     * The ActionTrigger must have either an onClick or an href field, or both.
+     */
+    actions?: Array<ActionTrigger>,
+
+    /**
+     * If present, dismiss button is on right side. If not, no button appears
+     */
+    onDismiss?: ?() => void,
+|};
+
+const colorForKind = (kind: BannerKind) => {
+    switch (kind) {
+        case "info":
+            return Color.blue;
+        case "success":
+            return Color.green;
+        case "warning":
+            return Color.gold;
+        case "critical":
+            return Color.red;
+        default:
+            return Color.blue;
+    }
+};
+
+// TODO(WB-1409): Use Phosphor icons instead of custom svgs. Also, use
+// Wonder Blocks Icon instead of img in the render fucntion.
+const iconForKind = (kind: BannerKind) => {
+    switch (kind) {
+        case "info":
+            return "/info.svg";
+        case "success":
+            return "/success.svg";
+        case "warning":
+            return "/warning.svg";
+        case "critical":
+            return "/critical.svg";
+        default:
+            return "/info.svg";
+    }
+};
+
+/**
+ * Banner. A banner displays a prominent message and related optional actions.
+ * It can be used as a way of informing the user of important changes.
+ * Typically, it is displayed toward the top of the screen.
+ *
+ * ### Usage
+ * ```jsx
+ * import Banner from "@khanacademy/wonder-blocks-banner";
+ *
+ * <Banner
+ *     text="Here is some example text."
+ *     kind="success"
+ *     layout="floating"
+ *     actions={[
+ *         {title: "Button 1", onClick: () => {}},
+ *         {title: "Button 2", onClick: () => {}},
+ *     ]}
+ *     onDismiss={() => {console.log("Has been dismissed.")}}
+ * />
+ * ```
+ */
+const Banner = (props: Props): React.Node => {
+    const {actions, onDismiss, kind, layout = "full-width", text} = props;
```

Comment:
I used `layout = "full-width"` to set "full-width" as the default value for `layout`. Should I use `type DefaultProps = {|layout: $PropertyType<Props, "full-width">|}` alongside a `static defaultProps: DefaultProps` like in other files? Would that work for functional components?

Reply:
For default props on functional component you should do something that looks like this:
```
const Foo = (props: Props): React.Node => {
    return null;
};

type DefaultProps = {|
   bar: string,
|};
Foo.defaultProps: DefaultProps = {bar: "hello"};

export {Foo};
```
This way when the component is used, flow make those props in `Foo.defaultProps` behave as optional props.

Reply:
I got it to work!